debezium-core/src/main/java/io/debezium/data/DebeziumDecimal.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.debezium.data;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.util.Optional;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.apache.kafka.connect.data.Struct;
+
+/**
+ * Extension of plain a {@link BigDecimal} type that adds support for new features
+ * like special values handling - NaN, infinity; 
+ *
+ * @author Jiri Pechanec
+ *
+ */
+public class DebeziumDecimal {
+    /**
+     * Special values for floating-point and numeric types
+     */
+    private static enum SpecialValue {
```

Comment:
Could you upper-case the enum constants?

Reply:
ok